Code/EnginePlugins/RecastPlugin/Components/DetourCrowdAgentComponent.cpp

```diff
@@ -0,0 +1,251 @@
+#include <RecastPlugin/RecastPluginPCH.h>
+
+#include <Core/WorldSerializer/WorldReader.h>
+#include <Core/WorldSerializer/WorldWriter.h>
+#include <GameEngine/Physics/CharacterControllerComponent.h>
+#include <RecastPlugin/Utils/RcMath.h>
+#include <RendererCore/Debug/DebugRenderer.h>
+#include <DetourCrowd.h>
+#include <RecastPlugin/WorldModule/DetourCrowdWorldModule.h>
+#include <RecastPlugin/Components/DetourCrowdAgentComponent.h>
+
+//////////////////////////////////////////////////////////////////////////
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezDetourCrowdAgentComponent, 1, ezComponentMode::Dynamic)
+{
+  EZ_BEGIN_ATTRIBUTES
+  {
+    new ezCategoryAttribute("AI/Recast"),
+  }
+  EZ_END_ATTRIBUTES;
+
+  EZ_BEGIN_PROPERTIES
+  {
+    EZ_MEMBER_PROPERTY("Radius",m_fRadius)->AddAttributes(new ezDefaultValueAttribute(0.3f)),
+    EZ_MEMBER_PROPERTY("Height",m_fHeight)->AddAttributes(new ezDefaultValueAttribute(1.8f)),
+    EZ_MEMBER_PROPERTY("MaxSpeed",m_fMaxSpeed)->AddAttributes(new ezDefaultValueAttribute(3.5f)),
+    EZ_MEMBER_PROPERTY("MaxAcceleration",m_fMaxAcceleration)->AddAttributes(new ezDefaultValueAttribute(10.0f)),
+    EZ_MEMBER_PROPERTY("StoppingDistance",m_fStoppingDistance)->AddAttributes(new ezDefaultValueAttribute(1.0f)),
+  }
+  EZ_END_PROPERTIES;
+}
+EZ_END_COMPONENT_TYPE
+// clang-format on
+
+ezDetourCrowdAgentComponent::ezDetourCrowdAgentComponent() = default;
+ezDetourCrowdAgentComponent::~ezDetourCrowdAgentComponent() = default;
+
+void ezDetourCrowdAgentComponent::SerializeComponent(ezWorldWriter& stream) const
+{
+  SUPER::SerializeComponent(stream);
+  ezStreamWriter& s = stream.GetStream();
+
+  s << m_fRadius;
+  s << m_fHeight;
+  s << m_fMaxSpeed;
+  s << m_fMaxAcceleration;
+  s << m_fStoppingDistance;
+}
+
+void ezDetourCrowdAgentComponent::DeserializeComponent(ezWorldReader& stream)
+{
+  SUPER::DeserializeComponent(stream);
+  ezStreamReader& s = stream.GetStream();
+
+  s >> m_fRadius;
+  s >> m_fHeight;
+  s >> m_fMaxSpeed;
+  s >> m_fMaxAcceleration;
+  s >> m_fStoppingDistance;
+}
+
+void ezDetourCrowdAgentComponent::FillAgentParams(ezDetourCrowdAgentParams& out_params) const
+{
+  out_params.m_fRadius = m_fRadius;
+  out_params.m_fHeight = m_fHeight;
+  out_params.m_fMaxSpeed = m_fMaxSpeed;
+  out_params.m_fMaxAcceleration = m_fMaxAcceleration;
+}
+
+void ezDetourCrowdAgentComponent::SetTargetPosition(const ezVec3& vPosition)
+{
+  m_vTargetPosition = vPosition;
+  m_PathToTargetState = ezAgentPathFindingState::HasTargetWaitingForPath;
+  m_bTargetDirty = true;
+}
+
+void ezDetourCrowdAgentComponent::ClearTargetPosition()
+{
+  m_PathToTargetState = ezAgentPathFindingState::HasNoTarget;
+}
+
+void ezDetourCrowdAgentComponent::OnActivated()
+{
+  SUPER::OnActivated();
+}
+
+void ezDetourCrowdAgentComponent::OnDeactivated()
+{
+  SUPER::OnDeactivated();
+}
```

Comment:
No need to override, if they don't do anything.

Reply:
Yep, that's a leftover, I'll remove it